src/main/java/com/mparticle/kits/PilgrimSDKKit.java

```diff
@@ -0,0 +1,195 @@
+        // can be null if it hasn't been set previously
+        if (info == null) {
+            info = new PilgrimUserInfo();
+        }
+        return info;
+    }
+
+    /* Section: User attribute set */
+    @Override
+    public void onSetAllUserAttributes(Map<String, String> attributes, Map<String, List<String>> attributeLists, FilteredMParticleUser filteredMParticleUser) {
+        // NOTE: attributeList not supported
+        // Is this ID correct?
+        updateUser(filteredMParticleUser.mpUser);
+        PilgrimUserInfo info = getUserInfo();
+        for (Map.Entry<String, String> entry : attributes.entrySet()) {
+            info.put(entry.getKey(), entry.getValue());
+        }
+        PilgrimSdk.get().setUserInfo(info);
+    }
+
+
+    @Override
+    public void onIncrementUserAttribute(String s, String s1, FilteredMParticleUser filteredMParticleUser) {
+        // Ignored,  Not supported atm
+    }
+
+    @Override
+    public void onRemoveUserAttribute(String s, FilteredMParticleUser filteredMParticleUser) {
+        PilgrimUserInfo info = getUserInfo();
+        // Do we want to log if it doesn't exist?
+        info.remove(s);
+    }
+
+    @Override
+    public void onSetUserAttribute(String s, Object o, FilteredMParticleUser filteredMParticleUser) {
+        PilgrimUserInfo info = getUserInfo();
+        info.put(s, o.toString());
+        // update
+        PilgrimSdk.get().setUserInfo(info);
+    }
+
+    @Override
+    public void onSetUserTag(String s, FilteredMParticleUser filteredMParticleUser) {
+        // Ignored, not supported atm.
+    }
+
+    @Override
+    public boolean supportsAttributeLists() {
+        // We don't support attribute lists
+        // the PilgrimSdk currently only
+        // allows single k => value attributes.
+        return false;
+    }
+
+    @Override
+    public void onSetUserAttributeList(String s, List<String> list, FilteredMParticleUser filteredMParticleUser) {
+        // Ignored, not supported
+    }
+
+    @Override
+    public void onConsentStateUpdated(ConsentState consentState, ConsentState consentState1, FilteredMParticleUser filteredMParticleUser) {
+        // Ignore?
+    }
+
+    @Override
+    public void onIdentifyCompleted(MParticleUser mParticleUser, FilteredIdentityApiRequest filteredIdentityApiRequest) {
+        updateUser(mParticleUser);
+    }
+
+    @Override
+    public void onLoginCompleted(MParticleUser mParticleUser, FilteredIdentityApiRequest filteredIdentityApiRequest) {
+        updateUser(mParticleUser);
+    }
+
+    @Override
+    public void onModifyCompleted(MParticleUser mParticleUser, FilteredIdentityApiRequest filteredIdentityApiRequest) {
+        updateUser(mParticleUser);
+    }
+
+    private void updateUser(MParticleUser mParticleUser) {
+        PilgrimUserInfo info = getUserInfo();
+        String customerId = mParticleUser.getUserIdentities().get(MParticle.IdentityType.CustomerId);
```

Comment:
Is `CustomerId` based on the user or the app? Like if I installed this in app `Hello` would this always return the same ID for my app named `Hello` or would it do it based on each device?

Reply:
Per docs and what they responded in their channel: it's based on the device, and the current local MParticleUser https://docs.mparticle.com/developers/sdk/android/idsync/

Should we just use the `String installId = PilgrimSdk.getInstallId();` as the userId?

Reply:
Actually after talking with the mParticle team, I'm now using their `getId()` accessor and transform it into a string, which is what they recommended for the time being.